kornia/geometry/transform/thin_plate_spline.py

```diff
@@ -0,0 +1,148 @@
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+
+from kornia.utils import _extract_device_dtype
+
+__all__ = [
+    "get_tps_parameters",
+    "warp_points_tps",
+    "warp_img_tensor_tps"
+]
+
+# utilities for computing thin plate spline transforms
+
+
+def _pair_square_euclidean(tensor1: torch.Tensor, tensor2: torch.Tensor) -> torch.Tensor:
+    """Compute the pairwise squared euclidean distance matrices (B, N, M) between two tensors
+        with shapes (B, N, C) and (B, M, C)."""
+    # ||t1-t2||^2 = (t1-t2)^T(t1-t2) = t1^T*t1 + t2^T*t2 - 2*t1^T*t2
+    t1_sq: torch.Tensor = tensor1.mul(tensor1).sum(dim=-1, keepdim=True)
+    t2_sq: torch.Tensor = tensor2.mul(tensor2).sum(dim=-1, keepdim=True).transpose(1, 2)
+    t1_t2: torch.Tensor = tensor1.matmul(tensor2.transpose(1, 2))
+    square_dist: torch.Tensor = -2 * t1_t2 + t1_sq + t2_sq
+    square_dist: torch.Tensor = square_dist.clamp(min=0)  # handle possible numerical errors
+    return square_dist
+
+
+def _kernel_distance(squared_distances: torch.Tensor, eps: float = 1e-8) -> torch.Tensor:
+    """Compute the TPS kernel distance function: r^2*log(r), where r is the euclidean distance.
+    Since log(r) = 1/2*log(r^2), this function takes the squared distance matrix and calculates
+    0.5 * r^2 * log(r^2)."""
+    # r^2 * log(r) = 1/2 * r^2 * log(r^2)
+    return 0.5 * squared_distances * squared_distances.add(eps).log()
+
+
+def get_tps_parameters(points_src: torch.Tensor, points_target: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
```

Comment:
I suggest a differents name according other functions we have - `get_tps_transform`. See https://kornia.readthedocs.io/en/latest/geometry.transform.html?highlight=get_parameters#kornia.geometry.transform.get_perspective_transform

For variable names `points_target` -> `points_dst`. We try to use `src` <-> `dst` convention.